README.md
# TYPO3 Extension "a21glossary"
A21 Glossary - automatical conversion of all abbreviations and acronyms in the special tags for accessibility issues

## Version 0.9.4




<table>
	<tr><td>Version</td><td>0.9.4 stable</td></tr>
	<tr><td>Release date</td><td>08. December 2011</td></tr>
	<tr><td>Author</td><td>Ronny Vorpahl</td></tr>
	<tr><td>Comment</td><td>added a second (more accessible) configuration, several bugfixes</td></tr>
</table>

## !! Attention !!
This is an **automatically** generated git version, based on the release into the [TYPO3 Extension Repository](http://www.typo3.org/extensions/).
You can find the original version at http://typo3.org/extensions/repository/view/a21glossary/ .

It does not make sense to make pull requests as this repository has been created automatically from 3rd party, not from the original author(s).

Every version of the extension is tagged with the version number, therefore you can switch quite easily between different versions.


-- never edit the line below (typo3-ter) --

ChangeLog

(add new changes on top of this file)

Please Note: Clear all Caches yourself after installing this extension

11-11-01 (0.8.4) Ronny Vorpahl <vorpahl@artplan21.de>

	* changed Label of tx_a21glossary_main.link as suggested in
		http://forge.typo3.org/issues/31006

	* committing Bugfix/Maintenance Release from Oliver Hader - Thank you very much by the way
		53539	2011-10-28 19:36	Oliver Hader	[TASK] Update ext_emconf.php
		53538	2011-10-28 19:33	Oliver Hader	[BUGFIX] #31388: Applying exclude tags is plain wrong
		53535	2011-10-28 18:51	Oliver Hader	[TASK] Add missing locallang.xml file
		53534	2011-10-28 18:42	Oliver Hader	[TASK] Update ext_emconf.php
		53533	2011-10-28 18:39	Oliver Hader	[TASK] Add missing comment constant for anchor solution
		53532	2011-10-28 18:37	Oliver Hader	[FEATURE] #31007: Integrate labels of types shown in frontend to locallang files
		53531	2011-10-28 18:12	Oliver Hader	[BUGFIX] #31009: Paragraphs in description are not rendered in frontend
		53530	2011-10-28 18:08	Oliver Hader	[BUGFIX] Constant storagePage can be a list
		53529	2011-10-28 18:03	Oliver Hader	[BUGFIX] #31386: Wrong RTE configuration in TCA
		53527	2011-10-28 17:44	Oliver Hader	[BUG] #31383: PHP warning if processed before caching

11-10-05 (0.8.3) Ronny Vorpahl <vorpahl@artplan21.de>

	* Moved three basic settings into extension config, still overridable by typoscript
		* glossaryWHEN
			* before Caching: faster, but non-caching content won't be affected (recommended)
			* after Caching: slower, but whole even uncached content will be affected
		* glossaryWHERE
			* whole content (old style)
			* use Typo3Search-Tags (recommended if using idexed_search)
			* i will set my own inclusions/exclusions using a21glossary-markers (see manual, they work in all modes)
		* excludeTags
	* updated documentation and README.txt (Debug Cheatsheet)
	* added clearCacheOnLoad = 1


11-10-04 (0.8.2) Ronny Vorpahl <vorpahl@artplan21.de>

	* major cleanup, partially rewrite of a21glossary algorithm


11-09-30 (0.8.1) Ronny Vorpahl <vorpahl@artplan21.de>

	* preg_quote and preg_replace is reguired now - no more workarounds
	* minor cleanup


11-08-29 (0.8.0) Benjamin Mack <benni@typo3.org>

	* Typo3 4.5 compatibility enhancement
	* code cleanup according to Typo3 Coding guidelines

	* behavioral change
		glossary starts without explicit inclusion tags: $this->splitAndReplace($document, 1);
		glossary will only start if inclusion settings are met: $this->splitAndReplace($document);


10-11-24 (0.7.3) Ronny Vorpahl <vorpahl@artplan21.de>

	* removed line break error


10-11-24 (0.7.2) Ronny Vorpahl <vorpahl@artplan21.de>

	* maintenance
		* removed ereg_replace, PREG_REPLACE IS NOW MANDATORY ('about time)
		* translated README.txt into german
		* improved debug:noncaching, regexp
		* debug marksers implies now noncaching
		* removed last window line breaks from several files


10-11-23 (0.7.1) Ronny Vorpahl <vorpahl@artplan21.de>

	* fixed double entry in excludeTags (appeared in 0.7.0), glossary now with now with double entry check

	* maintenance: debug commands for frontend usage
		* added debugQuery functions, use  &tx_a21glossary[query]=1
		* added debugResult functions, use  &tx_a21glossary[result]=1
		* added keepMarkers functions, use  &tx_a21glossary[keep]=1
		* show included and excluded page parts &tx_a21glossary[markers]=1
		* highlight replaced words &tx_a21glossary[highlight]=1
		* other commands (use like "highlight"):
			info, conf, noncaching,regexp,output,input

	* maintenance: other tasks
		* minor cleanup of comments
		* inval()'d many params
		* added README.txt with debug hints
		* TODO update manual, see README.TXT for new debug commands


10-10-05 (0.7.0) Ronny Vorpahl <vorpahl@artplan21.de>

	* MAJOR BEHAVIORAL CHANGE:
		* Now only caching content will be used for glossary replacement - glossary caches itself now MAJOR PERFORMANCE IMPROVMENT
		* Now using includeSearchTags=1 by default, thus <!--A21GLOSSARY_begin--><!--A21GLOSSARY_end--> and <!--A21GLOSSEX_begin--><!--A21GLOSSEX_end--> not neccessary by default
		* excludeTags-list is now extended by default to headers
	* please rethink if cached pages containing glossary links are no problem for you
	* this makes the a21_glossary compatible to 0.4-branch again, with slightly diffrent behavior of course


10-07-22 (0.6.4) Ronny Vorpahl <vorpahl@artplan21.de>

	* TODO recheck utf-8 compatibility - especially with russian chars
	* SELECT uid,short,longversion FROM tx_a21glossary_main WHERE longversion REGEXP '[^ -~öäüÖÄÜé]'
	* ERROR: Invalid multibyte sequence in argument in a21glossary/class.tx_a21glossary.php line 244

	* OK TOCHECK mbstring.internal_encoding	UTF-8
	* OK TOCHECK mbstring.http_output	pass

	TODO try ini_set('mbstring.internal_encoding', 'UTF-8');


10-01-15 (0.6.4) Ronny Vorpahl <vorpahl@artplan21.de>

	* glossary works only on predefined pages types (config.tx_a21glossary.typeList = 0)

09-03-05 (0.6.3) Ronny Vorpahl <vorpahl@artplan21.de>

	* major security fix: intval'd tx_a21glossary[uid] (for detail view) to prevent sql injections

08-12-22 (0.6.2) Ronny Vorpahl <vorpahl@artplan21.de>

	* minor bugfix: duplicate detection now spports shortcut field
		* update: fixed double replacement by putting replacements in the right order (ordinary first, different shortcut second)
	* minor code cleanup

07-12-13 (0.6.1) Stephan Jorek	<jorek@artplan21.de>

	* bugfix: makeInstance('tslib_cObj') didn't check tslib_cObj existance

07-11-09 (0.6.0) Ronny Vorpahl <vorpahl@artplan21.de>

	* default hook handling changed; glossary will be called by default on the final page containing all uncached elements
	* this hook can be disabled; glossary can be called by userfunc on custom objects
	use this userfunc on page object to restore old functionality of glossary (see setup.txt for example)
	* added lots of customizable functionality and rules - global and for every single element (see users manual)

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "a21glossary".
 *
 * Auto generated 30-09-2013 14:12
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
	'title' => 'A21 Glossary',
	'description' => 'A21 Glossary - automatical conversion of all abbreviations and acronyms in the special tags for accessibility issues',
	'category' => 'fe',
	'shy' => 0,
	'version' => '0.9.4',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'loadOrder' => '',
	'module' => '',
	'state' => 'stable',
	'uploadfolder' => 0,
	'createDirs' => 'uploads/tx_a21glossary/rte/',
	'modify_tables' => '',
	'clearcacheonload' => 1,
	'lockType' => '',
	'author' => 'Ronny Vorpahl',
	'author_email' => 'info@artplan21.de',
	'author_company' => 'Artplan21 GmbH. - www.artplan21.de',
	'CGLcompliance' => NULL,
	'CGLcompliance_note' => NULL,
	'constraints' => 
	array (
		'depends' => 
		array (
			'php' => '5.1.0-0.0.0',
			'typo3' => '4.1.0-0.0.0',
		),
		'conflicts' => 
		array (
		),
		'suggests' => 
		array (
		),
	),
);

?>